src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/SegmentTest.java

```diff
@@ -0,0 +1,24 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SegmentTest {
+
+    @Test
+    void testRemoveEmptyTrailingFields() {
+
+        final List<String> segmentValues = Arrays.asList("str1", "str2", "", " ", null);
+        final List<String> expected = Arrays.asList("str1", "str2");
```

Comment:
Can be simplified now:
```suggestion
        final List<String> segmentValues = List.of("str1", "str2", "", " ", null);
        final List<String> expected = List.of("str1", "str2");
```

Reply:
Unfortunately` List.of()` doesn't allow null values and i wanted to cover everything under the `StringUtils::isNotBlank` predicate

Reply:
Right, forgot about that!